greedy_algorithms/dijkstra.cpp

```diff
@@ -45,20 +61,26 @@ int minDistance(int mdist[], bool vset[], int V) {
     return minInd;
 }
 
-// Utility function to print distances
+/**
+ * Function to print minimum distances to vertices.
+ * @param dist[] array of distances to vertices
+ * @param int number of vertices
+ */
 void print(int dist[], int V) {
-    cout << "\nVertex  Distance" << endl;
+    std::cout << "\nVertex  Distance" << std::endl;
     for (int i = 0; i < V; i++) {
         if (dist[i] < INT_MAX)
-            cout << i << "\t" << dist[i] << endl;
+            std::cout << i << "\t" << dist[i] << std::endl;
         else
-            cout << i << "\tINF" << endl;
+            std::cout << i << "\tINF" << std::endl;
     }
 }
 
-// The main function that finds the shortest path from given source
-// to all other vertices using Dijkstra's Algorithm.It doesn't work on negative
-// weights
+/**
+ * The main function that finds the shortest path from given source
+ * to all other vertices using Dijkstra's Algorithm. It doesn't work on negative
+ * weights.
+*/
 void Dijkstra(Graph graph, int src) {
```

Comment:
passing class directly like this does not guarantee changes to the original passed class instance. The compiler may optimize by instantiating a local variable copy. Pass pointer reference and update the code accordingly
```cpp
void Dijkstra(Graph *graph, int src) {
```
But since the function is not making modifications to the original class, the correct way to pass it would be to pass a reference to constant class instance:
```suggestion
void Dijkstra(const Graph &graph, int src) {
```